simpeg/electromagnetics/natural_source/receivers.py

```diff
@@ -407,89 +690,421 @@ def eval(self, src, mesh, f, return_complex=False):  # noqa: A003
+    component : {'real', 'imag'}
+        Tipper data type. For the tipper element :math:`T_{ij}` specified by the `orientation`
+        input argument, the receiver can be set to compute the following:
+        - 'real': Real component of the tipper (unitless)
+        - 'imag': Imaginary component of the tipper (unitless)
+    locations_bs : (n_loc, n_dim) numpy.ndarray, optional
+        Locations for remote horizontal magnetic field measurements (e.g. base station).
+        Must be same shape as `locations`.
+    locations_e : (n_loc, n_dim) numpy.ndarray, optional
+        Deprecated property. To be removed in simpeg 0.23.0.
+    locations_h : (n_loc, n_dim) numpy.ndarray, optional
+        Deprecated property. To be removed in simpeg 0.23.0.
+    return_complex : bool, optional
+        If ``True`` the complex impedance is evaluation, and the `component` input
+        argument is ignored. Do not use for inversion!
+    """
+
+    def __init__(
+        self,
+        locations,
+        orientation="zx",
+        component="real",
+        locations_bs=None,
+        locations_e=None,
+        locations_h=None,
+        return_complex=False,
+    ):
+        self.orientation = orientation
+        self.component = component
+        self.return_complex = return_complex
+
+        # check if locations and locations_bs have been provided
+        if locations_bs is not None:
+            # check that locations are same size
+            if locations_bs.size == locations.size:
+                self._locations_bs = locations_bs
+            else:
+                raise ValueError("location_bs needs to be same size as locations")
+
+        else:
+            # check shape of locations
+            if isinstance(locations, list):
+                if len(locations) == 1:
+                    pass
+                elif len(locations) == 2:
+                    if locations[0].size != locations[1].size:
+                        raise ValueError(
+                            "location_bs needs to be same size as locations"
+                        )
+                else:
+                    raise ValueError("incorrect size of list, must be length of 1 or 2")
+
+                self._locations_bs = locations[0]
+                locations = locations[-1]
+            elif isinstance(locations, np.ndarray):
+                self._locations_bs = locations
+            else:
+                raise ValueError("locations need to be either a list or numpy array")
+
+        super().__init__(locations)
+
+        if (locations_e is not None) or (locations_h is not None):
+            warnings.warn(
+                (
+                    "'locations_e' and 'locations_h' are deprecated properties that are unused by the Tipper class.",
+                    "Receiver locations are set using 'locations' (and 'locations_bs').",
+                    "These property will be removed in simpeg v.0.23.0.",
+                ),
+                DeprecationWarning,
```

Comment:
A `FutureWarning` is more suited for this type of warnings. `DeprecationWarning`s are [ignored by default](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) and intended to other developers. But we want these warnings to reach users if they are passing deprecated attributes.
```suggestion
                FutureWarning,
```